test/src/unittests/tonal/test_pitchsaliencefunction.py

```diff
@@ -0,0 +1,394 @@
+        # Provide a single input peak with a unit magnitude at the reference frequency,
+        freq_speaks = [55]
+        mag_speaks = [1]
+        outputLength  = 600
+
+        calculatedPitchSalience = PitchSalienceFunction(magnitudeThreshold=0)(freq_speaks, mag_speaks)
+        # With Mag. Threshold  = 0 outputs are zero
+        self.assertEqual(len(calculatedPitchSalience), outputLength)       
+        self.assertEqualVector(calculatedPitchSalience, zeros(600))        
+        """ FIXME
+        The intention here is to test if some peaks are correctly discarded, however, this test does not achieve this goal because there is only one input spectral peak.    
+        The magnitudeThreshold parameter defines the maximum allowed difference from the highest peak in dBs, so the input list of peaks should be of different magnitudes to see it working.
+        For example, to make it simple, we can add a few peaks with lower magnitudes than 1 and use magnitudeThreshold = 0. Then only the peak with the magnitude 1 will be selected and the current expectedPitchSalience will be still a correct expected test output
+        """
+
+    def testTwoPeaksHarmonics(self):
+        # Provide a 2 input peaks with a unit magnitude and validate
+        # that PSF is different depending on numberHarmonics configuration.
+        freq_speaks = [55, 110]
+        mag_speaks = [1, 1]
+        outputLength  = 600
+         # Length of the non-zero values for this Pitch Salience = 11
+        # Pitch salience for numberHarmonics=1
+        expectedPitchSalienceH1 = [1.0000000e+00, 9.7552824e-01, 9.0450847e-01, 7.9389262e-01, 6.5450847e-01,
+            5.0000000e-01, 3.4549147e-01, 2.0610739e-01, 9.5491491e-02, 2.4471754e-02,
+            3.7493994e-33]
+
+         # Pitch salience for numberHarmonics=20
+        expectedPitchSalienceH20 = [1.8000000e+00, 1.7559509e+00, 1.6281152e+00, 1.4290068e+00, 1.1781152e+00,
+            8.9999998e-01, 6.2188464e-01, 3.7099332e-01, 1.7188469e-01, 4.4049159e-02,
+            6.7489186e-33]
+
+        calculatedPitchSalience1H = PitchSalienceFunction(numberHarmonics=1)(freq_speaks, mag_speaks)
+        self.assertEqual(len(calculatedPitchSalience1H), outputLength)
+        calculatedPitchSalience20H = PitchSalienceFunction(numberHarmonics=20)(freq_speaks, mag_speaks)
+        self.assertEqual(len(calculatedPitchSalience20H), outputLength)
+
+        if calculatedPitchSalience1H.tolist()!=calculatedPitchSalience20H.tolist():
+        #  Return a TRUE result
+           self.assertEqual(1,1)
+        else:
+        #  Return a FALSE result       
+           self.assertEqual(1,0)        
+
+        # Detailed contents check on returned values (regression check
+        self.assertAlmostEqualVectorFixedPrecision(calculatedPitchSalience1H[:11], expectedPitchSalienceH1, 7)
+        self.assertAlmostEqualVectorFixedPrecision(calculatedPitchSalience20H[:11], expectedPitchSalienceH20, 7)          
```

Comment:
Are the rest of the values zeros? Test comparing the entire 600-value vectors similarly to other tests. Otherwise, it is misleading, because it is not clear what happens to the rest of the values.